spec/vra_spec.rb

```diff
@@ -0,0 +1,417 @@
+    end
+  end
+
+  describe '#destroy' do
+    let(:resource_id)     { '8c1a833a-5844-4100-b58c-9cab3543c958' }
+    let(:state)           { { resource_id: resource_id } }
+    let(:vra_client)      { double('vra_client') }
+    let(:resources)       { double('resources') }
+    let(:destroy_request) { double('destroy_request') }
+    let(:resource) do
+      double('server1',
+             id: resource_id,
+             name: 'server1',
+             ip_addresses: [ '5.6.7.8' ],
+             vm?: true)
+    end
+
+    before do
+      allow(driver).to receive(:vra_client).and_return(vra_client)
+      allow(driver).to receive(:wait_for_request).with(destroy_request)
+      allow(vra_client).to receive(:resources).and_return(resources)
+      allow(resources).to receive(:by_id).and_return(resource)
+      allow(resource).to receive(:destroy).and_return(destroy_request)
+      allow(destroy_request).to receive(:id).and_return('6da65982-7c33-4e6e-b346-fdf4bcbf01ab')
+    end
+
+    context 'when the resource is not created' do
+      let(:state) { {} }
+      it 'does not look up the resource if no resource ID exists' do
+        expect(vra_client.resources).not_to receive(:by_id)
+        driver.destroy(state)
+      end
+    end
+
+    it 'looks up the resource record' do
+      expect(vra_client.resources).to receive(:by_id).with(resource_id).and_return(resource)
+      driver.destroy(state)
+    end
+
+    context 'when the resource record cannot be found' do
+      it 'does not raise an exception' do
+        allow(vra_client.resources).to receive(:by_id).with(resource_id).and_raise(Vra::Exception::NotFound)
+        expect { driver.destroy(state) }.not_to raise_error
+      end
+    end
+
+    describe 'creating the destroy request' do
+      context 'when the destroy method or server is not found' do
+        it 'does not raise an exception' do
+          allow(resource).to receive(:destroy).and_raise(Vra::Exception::NotFound)
+          expect { driver.destroy(state) }.not_to raise_error
+        end
+      end
+
+      it 'calls #destroy on the server' do
+        expect(resource).to receive(:destroy).and_return(destroy_request)
+        driver.destroy(state)
+      end
+    end
+
+    it 'waits for the destroy request to succeed' do
+      expect(driver).to receive(:wait_for_request).with(destroy_request)
+      driver.destroy(state)
+    end
+  end
+
+  describe '#catalog_request' do
+    let(:catalog_request) { double('catalog_request') }
+    let(:vra_client)      { double('vra_client') }
+    let(:catalog)         { double('catalog') }
+    before do
+      allow(driver).to receive(:vra_client).and_return(vra_client)
+      allow(vra_client).to receive(:catalog).and_return(catalog)
+      allow(catalog).to receive(:request).and_return(catalog_request)
+      [ :cpus=, :memory=, :requested_for=, :lease_days=, :notes=, :subtenant_id=, :set_parameter ].each do |method|
+        allow(catalog_request).to receive(method)
```

Comment:
Can you educate me on how to do that?  I don't see a way to set an attr_writer-style method via double(), only via allow/expect.